src/Control/Email/Email.php

```diff
@@ -3,766 +3,618 @@
-     * @return mixed Success of the sending operation from an MTA perspective. Doesn't actually give any indication if
-     * the mail has been delivered to the recipient properly). See Mailer->sendPlain() for return type details.
+     * @return string Debug info
      */
-    public function send($messageID = null)
+    public function debug()
     {
-        Requirements::clear();
-
-        $this->parseVariables();
+        $this->render();
 
-        if (empty($this->from)) {
-            $this->from = Email::config()->admin_email;
-        }
+        return "<h2>Email template {$this->class}:</h2>\n" . '<pre>' . $this->getSwiftMessage()->toString() . '</pre>';
+    }
 
-        $headers = $this->customHeaders;
+    /**
+     * @return string
+     */
+    public function getTemplate()
+    {
+        return $this->template;
+    }
 
-        if ($messageID) {
-            $headers['X-SilverStripeMessageID'] = project() . '.' . $messageID;
+    /**
+     * Set the template to render the email with
+     *
+     * @param string $template
+     * @return $this
+     */
+    public function setTemplate($template)
+    {
+        if (substr($template, -3) == '.ss') {
+            $template = substr($template, 0, -3);
         }
+        $this->template = $template;
 
-        if (project()) {
-            $headers['X-SilverStripeSite'] = project();
-        }
+        return $this;
+    }
 
+    /**
+     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
+     *
+     * @return bool
+     */
+    public function IsEmail()
+    {
+        return true;
+    }
 
-        $to = $this->to;
-        $from = $this->from;
-        $subject = $this->subject;
-        if ($sendAllTo = $this->config()->send_all_emails_to) {
-            $subject .= " [addressed to $to";
-            $to = $sendAllTo;
-            if ($this->cc) {
-                $subject .= ", cc to $this->cc";
-            }
-            if ($this->bcc) {
-                $subject .= ", bcc to $this->bcc";
-            }
-            $subject .= ']';
-            unset($headers['Cc']);
-            unset($headers['Bcc']);
-        } else {
-            if ($this->cc) {
-                $headers['Cc'] = $this->cc;
-            }
-            if ($this->bcc) {
-                $headers['Bcc'] = $this->bcc;
-            }
+    /**
+     * @param string $template
+     * @param array|ViewableData $data
+     * @param callable|null $callback
+     * @return static
+     */
+    public static function create_from_callback($template, $data, $callback = null)
+    {
+        // create our email message from the arguments
+        $message = static::create();
+        $message->setTemplate($template);
+        $message->setData($data);
+
+        // if a callback is provided, call it against the message
+        if (is_callable($callback)) {
+            $callback($message);
         }
 
+        $message->render();
```

Comment:
`render` is the replacement of `populateTemplate` which had to be called to "render" the email in v3.

So yes, devs do have to call this themselves if they don't use the callback method for creating emails, but this is not a change in behaviour IMO.

The problem with `send` calling render is what if a dev has manually set the body of the email? it'll be overridden and it'd be impossible to send a manually constructed email.